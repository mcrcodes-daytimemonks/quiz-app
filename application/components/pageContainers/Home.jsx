import Login from "../Login";
import Dashboard from "../Dashboard";
import css from "../../styles/Home.module.css";
import { useState, useEffect } from "react";
import Head from "next/head";

export default function Home() {
  const [cachedUsername, setCachedUsername] = useState(null);
  const [username, setUsername] = useState("");

  const handleUsernameChange = (e) => {
    setUsername(e.target.value);
  };

  const handleLogin = (e) => {
    e.preventDefault();
    localStorage.setItem("username", username);
    window.dispatchEvent(new Event("username"));
    setUsername("");
  };

  const handleLogout = () => {
    localStorage.setItem("username", "");
    window.dispatchEvent(new Event("username"));
  };

  useEffect(() => {
    const gotUsername = localStorage.getItem("username");
    if (gotUsername) {
      setCachedUsername(gotUsername);
    }
  }, []);

  useEffect(() => {
    window.addEventListener("username", () => {
      setCachedUsername(localStorage.getItem("username"));
    });

    return () =>
      window.removeEventListener("username", () =>
        setCachedUsername(localStorage.getItem("username"))
      );
  }, []);

  return (
    <div className={css.container}>
      <Head>
        <title>Quiz App: Home</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={css.main}>
        {cachedUsername ? (
          <Dashboard
            cachedUsername={cachedUsername}
            handleLogout={handleLogout}
          />
        ) : (
          <Login
            username={username}
            handleUsernameChange={handleUsernameChange}
            handleLogin={handleLogin}
          />
        )}
      </main>
    </div>
  );
}
