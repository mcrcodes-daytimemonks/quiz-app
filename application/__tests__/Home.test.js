import { render, screen } from "@testing-library/react";
import Home from "../pages/index";
import { SessionProvider } from "next-auth/react";

describe("Home", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <SessionProvider session={{ user: { name: "Jane Doe" } }}>
        <Home />
      </SessionProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
