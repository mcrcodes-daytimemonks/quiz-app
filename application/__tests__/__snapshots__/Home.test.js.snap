// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <div>
        Signed in as Jane Doe
        <button
          type="button"
        >
          Sign Out
        </button>
        <p>
          Hey, 
          <span>
            Jane!
          </span>
        </p>
        <form
          class="GameOptions__grid"
          name="game-options"
        >
          <fieldset
            class="categories"
          >
            <legend>
              Please choose a category
            </legend>
            <div
              class="category__input"
            >
              <input
                id="JavaScript"
                name="category"
                required=""
                type="radio"
                value="JavaScript"
              />
              <label
                class="radio-bttn-label"
                for="JavaScript"
              >
                JavaScript
              </label>
            </div>
            <div
              class="category__input"
            >
              <input
                id="HTML"
                name="category"
                required=""
                type="radio"
                value="HTML"
              />
              <label
                class="radio-bttn-label"
                for="HTML"
              >
                HTML
              </label>
            </div>
          </fieldset>
          <div>
            <label
              for="questionCount"
            >
              Number of questions: 
            </label>
            <input
              id="questionCount"
              max="5"
              min="1"
              required=""
              step="1"
              type="number"
              value="3"
            />
          </div>
          <button
            class="button primary"
            type="submit"
          >
            Play Game
          </button>
        </form>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
